tests/test_lsthing.py

```diff
@@ -702,6 +717,80 @@ def test_007_advanced_search_interactions(self):
                                        max_results=1000,
                                        combine_terms_with_and=True)
         assert len(results) == 0
+    
+    def test_008_validate_codevalue(self):
+        # Create project 1
+        name = str(uuid.uuid4())
+        status_1 = str(uuid.uuid4())
+        desc_1 = str(uuid.uuid4())
+        meta_dict = {
+            NAME_KEY: name,
+            IS_RESTRICTED_KEY: True,
+            STATUS_KEY: status_1,
+            START_DATE_KEY: datetime.now(),
+            DESCRIPTION_KEY: desc_1
+        }
+        error = None
+        try:
+            proj_1 = Project(recorded_by=self.client.username, **meta_dict)
+            proj_1.validate(self.client)
+        except ValueError as e:
+            error = e
+        self.assertEqual(str(error), f"Invalid 'code':'{status_1}' provided for the given 'code_type':'{PROJECT}' and 'code_kind':'{STATUS}'")
+    
+    def test_009_validate_ddicts(self):
+        # Create project 1
+        name = str(uuid.uuid4())
+        desc_1 = str(uuid.uuid4())
+        meta_dict = {
+            NAME_KEY: name,
+            IS_RESTRICTED_KEY: True,
+            START_DATE_KEY: datetime.now(),
+            DESCRIPTION_KEY: desc_1
+        }
+
+        proj_1 = Project(recorded_by=self.client.username, **meta_dict)
+        # set status to a CodeValue constructed with a DDict
+        STATUS_DDICT = ACASDDict(PROJECT, STATUS)
+        proj_1.metadata[PROJECT_METADATA][STATUS_KEY] = CodeValue(ACTIVE, ddict=STATUS_DDICT)
+        proj_1.validate(self.client)
+        assert True
+        # Now try setting status to an invalid CodeValue
+        status_1 = str(uuid.uuid4())
+        proj_1.metadata[PROJECT_METADATA][STATUS_KEY] = CodeValue(status_1, ddict=STATUS_DDICT)
+        error = None
+        try:
+            proj_1.validate(self.client)
+        except ValueError as e:
+            error = e
+        self.assertEqual(str(error), f"Invalid 'code':'{status_1}' provided for the given 'code_type':'{PROJECT}' and 'code_kind':'{STATUS}'")
+        # Now try adding a CodeValue that references an LsThing
+        proj_1.metadata[PROJECT_METADATA][STATUS_KEY] = CodeValue(ACTIVE, ddict=STATUS_DDICT)
+        proj_1.save(self.client)
```

Comment:
Thanks. This line is actually just setup (saving a new LsThing) so that we can reference it below. I've adopted your suggestion above when I first validate `proj_1` then again below when I validate `proj_2`